net/golioth/Kconfig

```diff
@@ -41,6 +41,7 @@ config GOLIOTH_FW
 
 config GOLIOTH_RPC
 	bool "Remote Procedure Call"
+	depends on QCBOR
```

Comment:
Looking once again into that... `select QCBOR` would probably be better. This is what we use with `GOLIOTH_FW` and `LOG_BACKEND_GOLIOTH`.

Reply:
Yeah, I chose to use `depends on` based on a quick read of the recommendations here: https://docs.zephyrproject.org/latest/build/kconfig/tips.html?highlight=kconfig%20tips#select-recommendations

> Avoid selecting symbols with prompts or dependencies. Prefer depends on. If depends on causes annoying bloat in configuration files, consider adding a Kconfig default for the most common value.

> Rare exceptions might include cases where you’re sure that the dependencies of the selecting and selected symbol will never drift out of sync, e.g. when dealing with two simple symbols defined close to one another within the same if.

> Common sense applies, but be aware that select often causes issues in practice. depends on is usually a cleaner and safer solution.

Reply:
Perhaps `select` is better in this case, since QCBOR doesn't have any dependencies?

Reply:
I'll go ahead and change to `select`, for consistency.

Reply:
>Perhaps select is better in this case, since QCBOR doesn't have any dependencies?

The dependency problem is a real thing. In Buildroot for example, which also is Kconfig based, if you select an option with dependencies (`depends on` entries), then all of those dependencies need to be duplicated (all `depends on` entries need to be copy-pasted) for the introduced option that does `select`. I haven't seen such practice with Zephyr, though. And that is not very developer-friendly.

> I'll go ahead and change to select, for consistency.

We can change `select` to `depends on` all at once if `select` will raise any issues. So yeah, consistency is the primary goal with my request here.